File: Ass1_Q3.py
n=int(input("Enter the number of terms to find sum "))
x=True
i=1
sum=1
pre=0
if(n<=0):
    print("enter positive integers only")
else:
    while(x==True ):
        sum=sum+1/(i+1)
        pre=pre+1/i
        if((sum-pre)<0.001)|(i==n):
            x=False
        else:
            i=i+1
    if(x==True):
        print("sum doesn't change by 0.001 and sum=" + str(sum))
    else:
        print("sum for " + str(n) + " terms=" +str(sum))

File: Ass1_Q3.cpp
#include <stdio.h>

int main() {
    double sum=1;
    double pre=0;
    double k=1;
    double d,summ;
    int n,m;
    printf("Integer input:");
    scanf("%d", &n);
    if(n>0){
        for(int i=1;sum-pre>0.001 && k==1;){
            if (i==n){k=0;}
            else{
                i=i+1;
                d=i;
                pre=sum;
                sum=sum+1/d;
            }
        }
        printf("%lf\n", sum);
    }
    else{
        m=-n;
        for(int i=1;sum-pre>0.001 && k==1;){
            if (i==m){k=0;}
            else{
                i=i+1;
                d=i;
                pre=sum;
                sum=sum+1/d;
            }
        }
        summ=-sum;
        printf("%lf\n", summ);
    }
    return 0;
}
